### ISSUE TITLE
fm => src

### ISSUE THREAD

**Issue by elephantux** [spatie/laravel-medialibrary#433]
Image format "src" don't work for me. 
```
    public function registerMediaConversions()
    {
        $this->addMediaConversion('index')
            ->setManipulations(['w' => 1000, 'h' => 560, 'fit' => 'max', 'fm' => 'src'])
            ->nonQueued();
        $this->addMediaConversion('thumb')
            ->setManipulations(['w' => 640, 'h' => 480, 'fit' => 'max', 'fm' => 'src'])
            ->nonQueued();
    }
```
and when save uploaded image:
```
        $model->addMedia($image->getRealPath())
            ->preservingOriginal()
            ->toCollection($title);
```
MediaLibrary can't retrieve my files extension (I try several files). In UploadFile class no extension method:
```
UploadedFile {#605 ▼
  -test: false
  -originalName: "cover.jpg"
  -mimeType: "image/jpeg"
  -size: 940996
  -error: 0
  path: "/tmp"
  filename: "phpNGqCzd"
  basename: "phpNGqCzd"
  pathname: "/tmp/phpNGqCzd"
  extension: ""
  realPath: "/tmp/phpNGqCzd"
  aTime: 2017-01-04 20:34:10
  mTime: 2017-01-04 20:34:10
  cTime: 2017-01-04 20:34:10
  inode: 654100
  size: 940996
  perms: 0100600
  owner: 900
  group: 900
  type: "file"
  writable: true
  readable: true
  executable: false
  file: true
  dir: false
  link: false
}
```

Why 'fm' => 'src' dosen't work for me? In MySQL filename saved without extension. In getUrl i got thumb.src image.
I use "spatie/laravel-medialibrary": "^4.10"

Thanks

**Comment by freekmurze** [comment]
`src` is not a valid image format. Try specifying `jpg` or `png`.

**Comment by elephantux** [comment]
This is from official docs:
>If you specify src in the fm-key, the derived image will have the same format as the original image.
https://docs.spatie.be/laravel-medialibrary/v4/converting-images/defining-conversions

And inside source code I can find this:
```
public function getResultExtension(string $originalFileExtension = '') : string
    {
        return array_reduce($this->getManipulations(), function ($carry, array $manipulation) {
            if (isset($manipulation['fm'])) {
                $keepExtensions = ['jpg', 'jpeg', 'png', 'gif'];

                return ($manipulation['fm'] === 'src' && in_array($carry, $keepExtensions)) ? $carry : $manipulation['fm'];
            }

            return $carry;
        }, $originalFileExtension);
    }
```

**Comment by freekmurze** [comment]
Ah sorry, totally forgot about the `src` option.

**Comment by freekmurze** [comment]
[This test](https://github.com/spatie/laravel-medialibrary/blob/485ae72/tests/FileAdder/IntegrationTest.php#L124-L135) seems to prove that the adding a file from an upload does work. 

I toyed around with the test a bit. Calling `getExtension()` on `$uploadedFile` does output `jpg`.

I'm guessing the problem isn't caused by the package but by how files are uploaded in your particular project or by some setting on your particular server.

**Comment by elephantux** [comment]
On production server (Digital Ocean LEMP) the same thing - any type of files saved with .src extension.

**Comment by freekmurze** [comment]
[This test](https://github.com/spatie/laravel-medialibrary/blob/485ae72/tests/Conversion/ConversionTest.php#L115) seems to prove that the `src` value for the `fm` conversion is working correctly.

Are you using the latest version of this package? Try to run the tests on your platform. Do they pass there?

**Comment by elephantux** [comment]
I using 4.10.2 version of Media Library.
Sorry but I never before run any tests. Can you please point me how to run this test on my platform?

**Comment by freekmurze** [comment]
1. Clone the package somewhere on your server
2. run `composer install` in that directory
3. execute `vendor/bin/phpunit` to run the tests

**Comment by elephantux** [comment]
![Image of Yaktocat](https://s27.postimg.org/jwd0d74ap/Capture.png)

**Comment by freekmurze** [comment]
Seems like the tests are running fine on your box. So still thinking that the problem must be caused specific to your project / server.

**Comment by sebastiandedeyne** [comment]
> In MySQL filename saved without extension.

I think this is where the issue's at. Could you post the data from the database row?

The extension is derived from the `UploadedFile`'s `originalName`, which seems to check out (it doesn't matter that `extension` is empty there), so not sure why the extension wouldn't be present in the database.

**Comment by elephantux** [comment]
When 'fm' => 'src' where is no extensions in DB:

![Image of Yaktocat](https://s27.postimg.org/lef410d41/Capture.png)

**Comment by elephantux** [comment]
I do not trim anything:
```
$data = $request->all();
$image = $data['image'];
$model->addMedia($image->getRealPath());
```

**Comment by Jasper9041** [comment]
I might be a little late to the party here, but since the issue is still open I might as well answer.

I was able to replicate your issue. Since you are calling `getRealPath()`, you are retreiving the actual real path to the uploaded file, which is a path to the /tmp directory. This path does not contain a file extension. See the 'realtpath' field in the included dump:
```
UploadedFile {#158 ▼
  -test: false
  -originalName: "dd.png"
  -mimeType: "image/png"
  -size: 217230
  -error: 0
  path: "/tmp"
  filename: "phpLvq1Df"
  basename: "phpLvq1Df"
  pathname: "/tmp/phpLvq1Df"
  extension: ""
  realPath: "/tmp/phpLvq1Df"
  aTime: 2017-02-02 15:00:34
  mTime: 2017-02-02 15:00:34
  cTime: 2017-02-02 15:00:34
  inode: 926336
  size: 217230
  perms: 0100600
  owner: 1000
  group: 1000
  type: "file"
  writable: true
  readable: true
  executable: false
  file: true
  dir: false
  link: false
}
```

This is why there is no extension in the the database. When converting with `'fm' => 'src`', laravel-medialibrary looks for the original file extension by calling `getExtensionAttribute()`, which does not have any extension to report. I included an extract of this method in Media.php:
```
public function getExtensionAttribute() : string
{
    return pathinfo($this->file_name, PATHINFO_EXTENSION);
}
```


@freekmurze Maybe the package needs to throw an exception if you try to add a file without extension to the library?

In conclusion: if you want to solve your issue you should be able to do this instead:
```
$data = $request->all();
$image = $data['image'];
$model->addMedia($image);
```
OR
```
$model->addMediaFromRequest('image');
```

**Comment by elephantux** [comment]
@Jasper9041 Thanks! It's work for me!
